src/main/java/com/atharvadholakia/password_manager/exception/GlobalExceptionHandler.java

```diff
@@ -29,22 +31,29 @@ public ResponseEntity<HashMap<String, String>> MethodArgsNotValidExceptionHandle
   }
 
   @ExceptionHandler(ResourceNotFoundException.class)
-  public ResponseEntity<HashMap<String, String>> ResourceNotFoundExceptionHandler(
+  public ResponseEntity<HashMap<String, String>> handleResourceNotFoundException(
       ResourceNotFoundException ex) {
     HashMap<String, String> response = new HashMap<>();
     response.put("error", ex.getMessage());
-
     return new ResponseEntity<>(response, HttpStatus.NOT_FOUND);
   }
 
-  @ExceptionHandler(Exception.class)
-  public ResponseEntity<HashMap<String, String>> GenericExceptionHandler(Exception ex) {
+  @ExceptionHandler(HttpMessageNotReadableException.class)
+  public ResponseEntity<HashMap<String, String>> handleHttpMessageNotReadableException(
+      HttpMessageNotReadableException ex) {
     HashMap<String, String> response = new HashMap<>();
-    response.put(
-        "error",
-        "We're sorry, but something went wrong on our end. Please try again later. If the"
-            + " problem persists, please contact support.");
-
-    return new ResponseEntity<>(response, HttpStatus.INTERNAL_SERVER_ERROR);
+    response.put("error", "Inputs can only be strings");
+    return new ResponseEntity<>(response, HttpStatus.BAD_REQUEST);
   }
+
+  // @ExceptionHandler(Exception.class)
```

Comment:
I think this should be uncommented, for 500 errors

Reply:
I have this commented when m debugging so that when a 500 error occurs I can understand whats casuing the error.